server/consumer.go

```diff
@@ -2251,12 +2251,17 @@ func (o *consumer) releaseAnyPendingRequests() {
 	if o.mset == nil || o.outq == nil || o.waiting.len() == 0 {
 		return
 	}
-	hdr := []byte("NATS/1.0 409 Consumer Deleted\r\n\r\n")
+	var hdr []byte
+	if !o.js.cluster.isConsumerAssigned(o.acc, o.stream, o.name) {
+		hdr = []byte("NATS/1.0 409 Consumer Deleted\r\n\r\n")
+	}
 	wq := o.waiting
 	o.waiting = nil
 	for i, rp := 0, wq.rp; i < wq.n; i++ {
 		if wr := wq.reqs[rp]; wr != nil {
-			o.outq.send(newJSPubMsg(wr.reply, _EMPTY_, _EMPTY_, hdr, nil, nil, 0))
+			if hdr != nil {
```

Comment:
If we are invalidating we need to send this to the end application, just with different hdr, like Consumer Peer Removed.

Reply:
That would be helpful and allow consumers on client side to notify users about pause in processing without raising doubts if consumer stopped working in scenarios when new leader election takes a lot of time for some reason.
However, we have to be aware of the fact, that when new leader sens a mesasge when it becomes one:

```
NATS/1.0 409 Leadership Change
```

That would mean, that in case of Peer Removal, clients will get two messages: 
First, that peer is removed (and there is no point in sending new fetch requests or counting idle hearbeats timeouts). Second, that a new leader has been elected (where clients know that all fetch requests has been discarded at it can start again).
**However: I do not want to add additional messaging to the clients in a fix, that would be released in a patch release.
I suggest splitting the work:**

This PR fixes sending `Consumer Deleted` and can go into next patch release.
Next PR will add `Consumer Peer Removed`, that would be part of 2.11

Reply:
They should only be sending leader change is what I meant. Only leaders hold requests, followers do not. So that is what I wanted to double check as to why that was not happening here.

Reply:
So we should not have these, but seems we do since we had folks report this..

Reply:
We first check if we are a leader, then we do stepdown follow up by this function, so it will always be called if a peer removed happens to be a consumer leader.

Thats why we need a check the assigmnent inside the function to make sure if it was a deleted consumer, or just a stepdown.

This is the context in which our function in question is called:
```go
	// Check if we are the leader and are being deleted.
	if dflag && o.isLeader() {
		// If we are clustered and node leader (probable from above), stepdown.
		if node := o.node; node != nil && node.Leader() {
			node.StepDown()
		}
		if advisory {
			o.sendDeleteAdvisoryLocked()
		}
		if o.isPullMode() {
			// Release any pending.
			o.releaseAnyPendingRequests()
		}
	}
```

Reply:
Which shows that this also sends delete advisory while it should not, right?

Reply:
So my question is how could we see these from user report?

Reply:
ok great, ready for final review you think?

Reply:
@derekcollison I think it's ready. Improved the code, added check for inproper advisories sent.